### PR TITLE
docs: add information on when not to use tree-shakable providers

### PR DESCRIPTION
**Pull request by sreevani-ship-it**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Does not have documentation for when not to use tree-shakable injector format.
Issue Number: 23621

## What is the new behavior?
Added a note explaining when not to use tree-shakable provider format.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by jenniferfell — APPROVED** [review]

**Comment by jenniferfell** [comment]
Looks okay to me. 

@StephenFluin Please review. If okay, approve and mark for merge. Thanks!

**Comment by StephenFluin** [comment]
@chembu This text is not correct. Any provider can be overridden, but they must be overriden using the traditional providers array syntax.

File: aio/content/guide/dependency-injection.md [hunk 1]
```diff
@@ -440,6 +440,12 @@ The service can be instantiated by configuring a factory function as shown below
 
 <code-example path="dependency-injection/src/app/tree-shaking/service.0.ts"  title="src/app/tree-shaking/service.0.ts" linenums="false"> </code-example>
 
+<div class="l-sub-section">
+
+To override tree-shakable providers, register the provider in the `providers[]` array of any Angular decorator that supports it.
```
Comment (trotyl) [inline aio/content/guide/dependency-injection.md]: What's `providers[]` array? Should be `providers` array for property name or `providers: []` array for usage or `Provider[]` for type annotation?

Comment (chembu) [reply aio/content/guide/dependency-injection.md]: Providers: [] array syntax. Updated note to clarify it.

**Review by StephenFluin — APPROVED** [review]

**Pull request closed without merging**